test/example.c

```diff
@@ -172,19 +181,19 @@ test_result test_compress(compr, comprLen, uncompr, uncomprLen)
 {
     int err;
     uLong len = (uLong)strlen(hello)+1;
-    test_result result;
 
     err = compress(compr, &comprLen, (const Bytef*)hello, len);
-    RETURN_ON_ERROR_WITH_MESSAGE(err, "compress", result);
+    RETURN_ON_ERROR_WITH_MESSAGE(err, "compress");
 
     strcpy((char*)uncompr, "garbage");
 
     err = uncompress(uncompr, &uncomprLen, compr, comprLen);
-    RETURN_ON_ERROR_WITH_MESSAGE(err, "uncompress", result);
+    RETURN_ON_ERROR_WITH_MESSAGE(err, "uncompress");
 
     if (strcmp((char*)uncompr, hello)) {
-        RETURN_WITH_MESSAGE("bad uncompress\n", result);
+        RETURN_WITH_MESSAGE("bad uncompress\n");
     } else {
+        test_result result;
```

Comment:
Nicer.  This happens quite a few times, though; want to add a RETURN_SUCCESS() to shorten the
common case?  I did that in https://github.com/dankegel/zlib/blob/tidier-maybe/test/example.c#L74

Reply:
Good idea. I'll do that.

In fact, looking at those cases makes me realize something. I could add 2 macros for the message / extended_message situation. Or I could just pass NULL as the extended message and use 1 macro.

This same idea could be applied to the existing macros.

So I'll do both of those things.